x-pack/plugins/beats/server/routes/api/register_create_enrollment_tokens_route.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import uuid from 'uuid';
+import {
+  get,
+  flatten
+} from 'lodash';
+import { INDEX_NAMES } from '../../../common/constants';
+import { callWithRequestFactory } from '../../lib/call_with_request_factory';
+import { wrapEsError } from '../../lib/error_wrappers';
+
+function persistTokens(callWithRequest, tokens) {
+  const body = flatten(tokens.map(token => [
+    { index: { _id: `enrollment_token:${token}` } },
+    { type: 'enrollment_token', enrollment_token: { token } }
+  ]));
+
+  const params = {
+    index: INDEX_NAMES.ADMIN,
+    type: '_doc',
+    body,
+    refresh: 'wait_for'
+  };
+
+  return callWithRequest('bulk', params);
+}
+
+// TODO: add license check pre-hook
+// TODO: write to Kibana audit log file
```

Comment:
I plan to add audit log functionality across all APIs that need it in a follow up PR.